enzy_htp/_interface/pymol_interface.py

```diff
@@ -101,6 +101,35 @@ def select_pymol_obj(self, pattern: str, pymol_obj_name: str,
 
         return sorted(result)
 
+    def point_mutate(self, pos_key: Tuple[str, int], 
+                     target: str, pymol_obj_name: str, 
+                     pymol_session: pymol2.PyMOL) -> None:
+        """
+        Performs a single point mutation on the PDB file in the PyMOL session in-place.
+        Args:
+            pos_key: the chain id and residue index.
+            target: the target residue name (3 letters).
+            pymol_obj_name: the name of the PDB file in PyMOL.
+            pymol_session: the current PyMOL session.
+        Returns:
+            None.
+        """
+
+        # load pymol wizard mutagenesis function
+        pymol_session.cmd.wizard("mutagenesis")
+        pymol_session.cmd.do("refresh_wizard")
+        
+        # select res idx to mutate to target
+        pymol_session.cmd.get_wizard().set_mode(target)
+        if pymol_obj_name == "":
+            pymol_session.cmd.get_wizard().do_select(pos_key[0] + "/" + str(pos_key[1]) + "/")
+        else:
+            pymol_session.cmd.get_wizard().do_select("/" + pymol_obj_name + "//" + pos_key[0] + "/" + str(pos_key[1]) + "/")
+
+        # select the best rotamer (defaulted to frame 1) and apply
+        pymol_session.cmd.frame(1)
```

Comment:
IMPORTANT:
It seems the lowest conformer is not frame 1 lol.
I tested using the unitest you wrote and I compared the structure and found frame 1 is just frame 1 and the best one should be frame 2 in this case. So we prob still need to determine it. (Maybe try to find something like current frame. or try just apply instead of choosing the frame. the most ideal will be obtaining the complete information from pymol)

Reply:
Green is WT. Blue is from enzy_htp using this function. White is from current pymol session.
Best frame:
<img width="357" alt="image" src="https://github.com/ChemBioHTP/EnzyHTP/assets/36098720/acab0a10-040b-4c2b-8c5e-f5437069fbeb">
Frame 1:
<img width="341" alt="image" src="https://github.com/ChemBioHTP/EnzyHTP/assets/36098720/4ff0bda4-516c-4d9d-a281-c0abaca3a58c">

Reply:
![image](https://github.com/ChemBioHTP/EnzyHTP/assets/68860396/49721d43-8a17-4279-9f4e-759522bc533f)

It seems as though the PyMOL in python has a different rotamer library, as it looks like the mutagenesis is picking frame 6 from 9 rather than frame 2 from 8. This is without cmd.frame(1). Green is WT, Blue is using enzy_htp, white is current session.

Frame 1:
![image](https://github.com/ChemBioHTP/EnzyHTP/assets/68860396/ae1e1d1b-cc65-4624-baf1-898a2dfd4b06)

Frame 2 (best frame):
![image](https://github.com/ChemBioHTP/EnzyHTP/assets/68860396/2ab5ab10-8b25-4493-a3a4-696de7a0fa1e)

It doesn't seem like it's explicitly choosing frame 1 anymore, which is a plus.

Reply:
I checked out the MD for this mutant and it is indeed "frame 2" in GUI is the most frequent conformer.

Reply:
This could be a reference https://pymolwiki.org/index.php/Rotkit

Reply:
Looks like PyMOL (and Rotkit) considers frame 1 as the most probable mutation based off the percentage shown below:

![image](https://github.com/ChemBioHTP/EnzyHTP/assets/68860396/0d5619b0-6700-4802-a4ec-6521c58fe911)

The "31.9%" next to "mutation" is the probability of the mutation actually occurring with this rotamer. The percentages are listed in decreasing order as the frames go on, which is why it defaults to frame "1" I think.

Reply:
![image](https://github.com/ChemBioHTP/EnzyHTP/assets/68860396/d17c1572-1c33-482d-afeb-c7cae2672ffe)

So I've found a way to print them all out. It seems like in this library, rotamer 6 (the one defaulted to when removing "frame(1)") has the lowest strain, leading me to believe the mutagenesis wizard defaults to the lowest strain mutation and applies it. If this is true, no more needs to be done, although the more common rotamer you found using MD won't be chosen due to the fact that the rotamer library isn't fully updated with the GUI for whatever reason.

Reply:
Wow great job!
I noticed the percentages too. I searched about it but didn't find any explanation. Frame 1 clearly have multiple clashes with a super high strain score comparing to others. Could it be the percentage of the equivalent rotamers? (i.e.: the permutation of all possible dihedrals have most of them corresponding to this rotamer)

Reply:
Great job finding the way to print out the scores again!
It will be great if we can have an argument to print those scores out and save the corresponding structures for debugging purposes.
For doing mutation, we can keep the current strategy of just removing frame(1).
For the inconsistency between pymol2 and pymol-GUI, we should open an issue in their [GitHub repo](https://github.com/schrodinger/pymol-open-source/issues). This is a good chance to practice writing an issue on GitHub. The key point is not to involve too much about EnzyHTP, isolate the problem itself and report to them.

Reply:
Ok, I just added a "debug" parameter to point_mutate() that does this

Reply:
Good job! Please let me know if you opened an issue for open-source-pymol and I will close this comment after that.

Reply:
I just opened an issue!